Assignment/SampleData.cs

```diff
@@ -1,30 +1,130 @@
 ﻿using System;
+using System.IO;
+using System.Linq;
 using System.Collections.Generic;
 
+#pragma warning disable CA1065
+
 namespace Assignment
 {
     public class SampleData : ISampleData
     {
+        private string? _FileName;

```

Comment:
There is no `set` or `get` logic, and the value is never changed once set in the constructor. A readonly autoprop makes the most sense for that situation.

```suggestion
        private string FileName { get; };
```